pkgs/development/libraries/appmenu-gtk-module/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv, fetchFromGitLab, fetchpatch, meson, cmake, pkg-config, systemd, gtk-doc, docbook-xsl-nons, ninja, glib, gtk2, gtk3 }:
+
+stdenv.mkDerivation {
+  pname = "appmenu-gtk-module";
+  version = "0.7.3.2";
+
+  outputs = [ "out" "devdoc" ];
+
+  src = fetchFromGitLab {
+    owner = "vala-panel-project";
+    repo = "vala-panel-appmenu";
+    rev = "$version";
+    sha256 = "sha256:0j6jm34gp1slxvqhwk3r9zngrf73w4vp5s7jv0bj5xh0ns49x81d";
+  };
+#  sourceRoot = "source/subprojects/appmenu-gtk-module";
+  nativeBuildInputs = [ meson cmake pkg-config systemd gtk-doc docbook-xsl-nons ninja ];
+  buildInputs = [ glib gtk2 gtk3 ];
```

Comment:
Hmm, what is it that requires gtk2? The [readme](https://gitlab.com/vala-panel-project/vala-panel-appmenu/-/blob/master/README.md) claims that the dependencies are


-    GLib (>= 2.50.0)
-    GTK+ (>= 3.22.0)
-    valac (>= 0.24.0)
-    libbamf (>=0.5.0)

so I imagine it should be sufficient with gtk3?

Reply:
Some packages like claws mail are still gtk2-only. gtk2 support is needed for appmenu to work with these programs.

Reply:
@LouisDK1 I think you need to make a `appmenu-gtk3-module` and a `appmenu-gtk2-module` to avoid the above issue mentioned. This is done with a lot of other packages

Reply:
@worldofpeace Good idea. Will look into that.

The comment below was written before I saw your reply.